src/lib/markbind/src/parser.js

```diff
@@ -95,17 +97,40 @@ class Parser {
     $('variable').each(function () {
       const variableElement = $(this);
       const variableName = variableElement.attr('name');
-      if (!variableName) {
-        // eslint-disable-next-line no-console
-        console.warn(`Missing 'name' for variable in ${fileName}\n`);
-        return;
-      }
-      if (!pageVariables[variableName]) {
-        const variableValue = nunjuckUtils.renderEscaped(nunjucks, md.renderInline(variableElement.html()), {
-          ...importedVariables, ...pageVariables, ...userDefinedVariables, ...includedVariables,
-        });
-        pageVariables[variableName] = variableValue;
-        Parser.VARIABLE_LOOKUP.get(fileName).set(variableName, variableValue);
+      const variableSource = $(this).attr('from');
+      if (variableSource !== undefined) {
+        try {
+          const variableFilePath = path.resolve(fileDir, variableSource);
+          const jsonData = fs.readFileSync(variableFilePath);
+          const varData = JSON.parse(jsonData);
+          Object.entries(varData).forEach((variableEntry) => {
+            const [varName, varValue] = variableEntry;
```

Comment:
There seems to be some repetition between the json and non-json cases. Should we abstract this out into a separate function? That will make what we are trying to do a little more explicit too.